src/personas/data_manager/tests.rs

```diff
@@ -730,33 +726,35 @@ fn setup_mdata_refresh<R: Rng>(data: &MutableData,
                                -> (RoutingNode, DataManager, Vec<XorName>) {
     let (mut new_node, mut new_dm, other_node_names) = setup_churn(rng);
 
-    let fragments = vec![FragmentInfo::MutableData(*data.name(), data.tag())];
-    let refresh_payload = unwrap!(serialise(&fragments));
+    let refresh = vec![Refresh::Chunk(data.id())];
+    let refresh = unwrap!(serialise(&refresh));
 
     for name in &other_node_names {
-        unwrap!(new_dm.handle_refresh(&mut new_node, *name, &refresh_payload));
+        unwrap!(new_dm.handle_refresh(&mut new_node, *name, &refresh));
     }
 
     (new_node, new_dm, other_node_names)
 }
 
-// Removes and returns `GetMData` request sent to the given destination authority.
-fn take_get_mdata_request(node: &mut RoutingNode,
-                          dst: &Authority<XorName>)
-                          -> (MutableDataId, Authority<XorName>) {
-    let message = take_request(node, |message| message.dst == *dst);
+// Removes and returns sent `GetMData` request.
+fn take_get_mdata_request(node: &mut RoutingNode) -> (MessageId, MutableDataId) {
+    let (msg_id, message) = take_request(node, |message| match message.request {
+        Request::GetMData { .. } => true,
+        _ => false,
+    });
+
     let (name, tag) =
         assert_match!(message.request, Request::GetMData { name, tag, .. } => (name, tag));
-    (MutableDataId(name, tag), message.src)
+    (msg_id, MutableDataId(name, tag))
 }
 
-fn take_request<F>(node: &mut RoutingNode, mut f: F) -> (RequestWrapper)
+fn take_request<F>(node: &mut RoutingNode, mut f: F) -> (MessageId, RequestWrapper)
```

Comment:
It would be helpful to have a comment that explains that `f` is a filter function here.